underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -261,7 +273,92 @@ public boolean mkdirs(String path, boolean createParent) throws IOException {
 
   @Override
   public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
-    return true;
+    if (path == null) {
+      return false;
+    }
+    if (isDirectory(path)) {
+      return true;
+    }
+    if (isFile(path)) {
+      LOG.error("Cannot create directory {} because it is already a file.", path);
+      return false;
+    }
+    if (!options.getCreateParent()) {
+      if (parentExists(path)) {
+        return mkdirsInternal(path);
+      } else {
+        LOG.error("Cannot create directory {} because parent does not exist", path);
+        return false;
+      }
+    }
+    // Parent directories should be created
+    if (parentExists(path)) {
+      return mkdirsInternal(path);
+    } else {
+      String parentKey = getParentKey(path);
+      // Recursively make the parent folders
+      return mkdirs(parentKey, true) && mkdirsInternal(path);
+    }
+  }
+
+  /**
+   * Creates a directory flagged file with the folder suffix.
+   *
+   * @param path the path to create a folder
+   * @return true if the operation was successful, false otherwise
+   */
+  private boolean mkdirsInternal(String path) {
+    try {
+      String keyAsFolder = makeQualifiedPath(stripPrefixIfPresent(path, Constants.HEADER_SWIFT));
+      SwiftOutputStream out = SwiftDirectClient.put(mAccess, keyAsFolder);
+      out.close();
+      return true;
+    } catch (IOException e) {
+      LOG.error("Failed to create directory: {}", path, e);
+      return false;
+    }
+  }
+
+  /**
+   * Treating Swift as a file system, checks if the parent directory exists.
+   *
+   * @param path the path to check
+   * @return true if the parent exists or if the path is root, false otherwise
+   */
+  private boolean parentExists(String path) throws IOException {
+    // Assume root always has a parent
```

Comment:
I think this assumption is in contradiction with `getParentKey` logic. I think it is better to assume the caller does this check so that the value returned `getParentKey` is non-null iff `parentExists` returns true. Makes sense?